openmdao/core/component.py

```diff
@@ -615,12 +586,110 @@ def declare_partials(self, of, wrt, dependent=True,
+                - 'fd': Finite Difference, 'cs': Complex Step
+        **kwargs : dict
+            Keyword arguments for controlling the behavior of the approximation.
         """
-        # If only one of rows/cols is specified
-        if (rows is None) ^ (cols is None):
-            raise ValueError('If one of rows/cols is specified, then both must be specified')
+        supported_methods = {'fd': FiniteDifference,
+                             'cs': ComplexStep,
+                             'exact': None}
+
+        if method not in supported_methods:
+            msg = 'Method "{}" is not supported, method must be one of {}'
+            raise ValueError(msg.format(method, supported_methods.keys()))
 
-        self._declared_partials.append((of, wrt, dependent, rows, cols, val))
+        # Analytic Derivative for this jacobian pair
+        if method == 'exact':
+
+            # If only one of rows/cols is specified
+            if (rows is None) ^ (cols is None):
+                raise ValueError('If one of rows/cols is specified, then both must be specified')
+
+            self._declared_partials.append((of, wrt, dependent, rows, cols, val))
+
+        # Approximation of the derivative, former API call approx_partials.
+        else:
+
+            if method not in self._approx_schemes:
+                self._approx_schemes[method] = supported_methods[method]()
+
+            # If rows/cols is specified
+            if rows is not None or cols is not None:
+                raise ValueError('Sparse FD specification not supported yet.')
+
+            # Need to declare the Jacobian element too.
+            self._declared_partials.append((of, wrt, True, rows, cols, val))
+
+            self._approximated_partials.append((of, wrt, method, kwargs))
+
+    def set_check_partial_options(self, wrt, method='fd', form=None, step=None, step_calc=None):
+        """
+        Set options that will be used for checking partial derivatives.
+
+        Parameters
+        ----------
+        wrt : str or list of str
+            The name or names of the variables that derivatives are taken with respect to.
+            This can contain the name of any input or output variable.
+            May also contain a glob pattern.
+        method : str
+            Method for check: "fd" for finite difference, "cs" for complex step.
+        form : str
+            Finite difference form for check, can be "forward", "central", or "backward". Leave
+            undeclared to keep unchanged from previous or default value.
+        step : float
+            Step size for finite difference check. Leave undeclared to keep unchanged from previous
+            or default value.
+        step_calc : str
+            Type of step calculation for check, can be "abs" for absolute (default) or "rel" for
+            relative.  Leave undeclared to keep unchanged from previous or default value.
+        """
+        supported_methods = {'fd': FiniteDifference,
```

Comment:
no need for a dict here because you don't use the values. Could be a set, or since it's so short, maybe just a tuple.

Reply:
fixed